### PR TITLE
feat: add support for Cloudflare Turnstile response param in signUp (#155)

### LINKED ISSUE #155: nhost_auth_dart package doesn't support Cloudflare Turntile bot protection
September's release introduced support of  [Cloudflare's Turnstile](https://www.cloudflare.com/en-gb/products/turnstile/) service to verify that the user is not a bot.

When this feature is enabled at the backend we have to add the `x-cf-turnstile-response` header to the ` POST /signup/email-password` endpoint.

### PR DESCRIPTION
**Pull request by wildsurfer**
Fixes #155

### CODE REVIEW

File: packages/nhost_auth_dart/lib/src/auth_client.dart [hunk 1]
```diff
@@ -220,7 +227,6 @@ class NhostAuthClient implements HasuraAuthClient {
       rethrow;
     }
   }
-
```
Comment (dbarrosop) [inline packages/nhost_auth_dart/lib/src/auth_client.dart:223]: can you add this newline back, please?

**Comment by dbarrosop** [comment]
Thanks for the PR, just a minor comment, otherwise, LGTM :)

**Comment by wildsurfer** [comment]
@dbarrosop I implemented your recommendations, let me know if you wnat anything else please

**Comment by dbarrosop** [comment]
Could you allow edits so I can fix the CI?

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by wildsurfer** [comment]
@dbarrosop 🤔 It was already enabled...
Screenshot 2024-12-13 at 11 47 59

image

**Comment by dbarrosop** [comment]
I will never understand github...

```
git push git@github.com:wildsurfer/nhost-dart.git wildsurfer-main
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Delta compression using up to 14 threads
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 528 bytes | 176.00 KiB/s, done.
Total 5 (delta 3), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To github.com:wildsurfer/nhost-dart.git
 ! [remote rejected] wildsurfer-main -> wildsurfer-main (permission denied)
error: failed to push some refs to 'github.com:wildsurfer/nhost-dart.git'
```

anyway, it is a simple fix so maybe it is faster if you do it yourself, edit the file `.github/workflows/test.nhost_auth_dart.yaml` and apply the following patch:

```diff
diff --git a/.github/workflows/test.nhost_auth_dart.yaml b/.github/workflows/test.nhost_auth_dart.yaml
index afacf73..8ad5a5f 100644
--- a/.github/workflows/test.nhost_auth_dart.yaml
+++ b/.github/workflows/test.nhost_auth_dart.yaml
@@ -22,7 +22,8 @@ on:
 # Run the quality job at 4am every day
 # - cron: '0 4 * * *'

-env: {}
+env:
+  MAX_PANA_MISSING_POINTS: 40
 jobs:
   # Ensure that the package meets a suitably high pub.dev score
   score-package-quality:
```

**Comment by wildsurfer** [comment]
@dbarrosop patch applied 🙂

**Comment by dbarrosop** [comment]
Thanks! Merging.

**Pull request merged**